frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-topology/utils.ts

```diff
@@ -333,6 +363,225 @@ export const getTopologyNodesEdges = (
   return { nodes, edges };
 };
 
+export const getTextWidth = (text: string, font: string = '0.8rem RedHatText'): number => {
+  if (!text || text.length === 0) {
+    return 0;
+  }
+  const canvas = document.createElement('canvas');
+  const context = canvas.getContext('2d');
+  if (!context) {
+    return text.length;
+  }
+  context.font = font;
+  const { width } = context.measureText(text);
+  return width;
+};
+
+export const extractDepsFromContextVariables = (contextVariable: string) => {
+  const regex = /(?:(?:\$\(tasks.))([a-z0-9_-]+)(?:.results+)(?:[.^\w]+\))/g;
+  let matches;
+  const deps = [];
+  // eslint-disable-next-line no-cond-assign
+  while ((matches = regex.exec(contextVariable)) !== null) {
+    // This is necessary to avoid infinite loops with zero-width matches
+    if (matches.index === regex.lastIndex) {
+      regex.lastIndex++;
+    }
+    if (matches) {
+      if (!deps.includes(matches[1])) {
+        deps.push(matches[1]);
+      }
+    }
+  }
+  return deps;
+};
+
+export const getSpacerNode = (node: PipelineMixedNodeModel): PipelineMixedNodeModel => ({
+  ...node,
+  height: 1,
+  width: 1,
+});
+
+export const getWhenStatus = (status: RunStatus): string => {
+  switch (status) {
+    case RunStatus.Succeeded:
+    case RunStatus.Failed:
+      return WhenStatus.Met;
+    case RunStatus.Skipped:
+    case RunStatus['In Progress']:
+    case RunStatus.Idle:
+      return WhenStatus.Unmet;
+    default:
+      return '';
+  }
+};
+
+export const taskWhenStatus = (task: PipelineTaskWithStatus) => {
+  if (!task.when) {
+    return undefined;
+  }
+  return getWhenStatus(task.status?.reason);
+};
```

Comment:
Should we use get here as well, similar to most other getters in this file?

```suggestion
export const `getTaskWhenStatus = (task: PipelineTaskWithStatus): WhenStatus => {
  if (!task.when) {
    return undefined;
  }
  return getWhenStatus(task.status?.reason);
};
```